src/OrleansRuntime/GrainDirectory/RemoteGrainDirectory.cs

```diff
@@ -336,24 +75,24 @@ public Task RegisterManySingleActivation(List<ActivationAddress> addresses, int
                 if (curGen == tuple.Item2 || curGen == GrainInfo.NO_ETAG)
                 {
                     // the grain entry either does not exist in the local partition (curGen = -1) or has not been updated
-                    result.Add(new Tuple<GrainId, int, List<Tuple<SiloAddress, ActivationId>>>(tuple.Item1, curGen, null));
+                    result.Add(new Tuple<GrainId, int, List<ActivationAddress>>(tuple.Item1, curGen, null));
                 }
                 else
                 {
                     // the grain entry has been updated -- fetch and return its current version
-                    Tuple<List<Tuple<SiloAddress, ActivationId>>, int> lookupResult = partition.LookUpGrain(tuple.Item1);
+                    var lookupResult = await router.LookupAsync(tuple.Item1, 1);
```

Comment:
Looks like a very different method now. It used to look up only locally, now it does that remotely. So the semantics changed. I cannot express any opinion about that, without full semantic understanding of what is going on in the new implementation.